mirroring/handler.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	handler := LogHandler{}
+	metadata, err := LoadTreeMetadata()
+	if err != nil {
+		return handler, err
+	}
+	handler.client, err = NewClient()
+	if err != nil {
+		return handler, err
+	}
+	handler.metadata = metadata
+	// stored information will contain duplicates. should these duplicates be checked?
+	/*
+			logRoot := trilliantypes.LogRootV1{}
+			logRootBytes, err := base64.StdEncoding.DecodeString(metadata.LogInfo.SignedTreeHead.LogRoot.String())
+			if err != nil {
+				return handler, err
+			}
+			err = logRoot.UnmarshalBinary(logRootBytes)
+			if err != nil {
+				return handler, err
+			}
+		logRoot.
+			if logRoot.RootHash != []byte(*metadata.LogInfo.RootHash)
+	*/
+	handler.newLeavesBuffer = make([]Artifact, 0)
+	return handler, nil
+}
+
+func (h *LogHandler) GetLeafBuffer() []Artifact {
+	return h.newLeavesBuffer
+}
+
+func (h *LogHandler) SetLeafBuffer(b []Artifact) {
+	h.newLeavesBuffer = b
+}
+
+func (h *LogHandler) GetLocalPublicKey() string {
+	return h.metadata.PublicKey
+}
+
+func (h *LogHandler) GetLocalTreeSize() int64 {
+	return h.metadata.SavedMaxIndex + 1
+}
+
+func (h *LogHandler) GetRemoteTreeSize() (int64, error) {
+	a, err := GetLogInfo()
+	if err != nil {
+		return 0, err
+	}
+	return *a.TreeSize, nil
+}
+
+func (h *LogHandler) GetRemoteRootSignature() (strfmt.Base64, error) {
+	a, err := GetLogInfo()
+	if err != nil {
+		return nil, err
+	}
+	return *a.SignedTreeHead.Signature, nil
+}
+
+func (h *LogHandler) GetRemoteRootPublicKey() (string, error) {
+	keyResp, err := h.client.Tlog.GetPublicKey(nil)
+	if err != nil {
+		return "", err
+	}
+	publicKey := keyResp.Payload
+
+	return publicKey, nil
+}
+
+func (h *LogHandler) GetRootSignature() ([]byte, error) {
+	s, err := base64.StdEncoding.DecodeString(h.metadata.LogInfo.SignedTreeHead.Signature.String())
+	if err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+func (h *LogHandler) GetLocalRootHash() string {
+	return *h.metadata.LogInfo.RootHash
+}
+
+func (h *LogHandler) GetRemoteRootHash() ([]byte, error) {
+	info, err := GetLogInfo()
+	if err != nil {
+		return nil, err
+	}
+	logRootBytes, err := base64.StdEncoding.DecodeString(info.SignedTreeHead.LogRoot.String())
+	if err != nil {
+		return nil, err
+	}
+	logRoot := trilliantypes.LogRootV1{}
+	err = logRoot.UnmarshalBinary(logRootBytes)
+	if err != nil {
+		return nil, err
+	}
+	return logRoot.RootHash, nil
+}
+
+func (h *LogHandler) SetRootHash(rootHash string) {
+	*h.metadata.LogInfo.RootHash = rootHash
+}
+
+func (h *LogHandler) SetPublicKey(publicKey string) {
+	h.metadata.PublicKey = publicKey
+}
+
+func (h *LogHandler) SetRootSignature(sig strfmt.Base64) {
+	*h.metadata.LogInfo.SignedTreeHead.Signature = sig
+}
+
+func (h *LogHandler) SetLocalTreeSize(treeSize int64) {
+	h.metadata.SavedMaxIndex = treeSize - 1
+}
+
+// if kind=="", get all kinds of leaves
+func (h *LogHandler) GetAllLeavesForKind(kind string) error {
+	str := viper.GetString("tree_file_dir")
```

Comment:
I think we could call this variable something more descriptive. What about `tree_file_dir_path` ? :P

Reply:
That makes sense, I renamed every '[prefix]_dir' key to '[prefix]_directory'